File: apps/agents/src/web-search/state.ts
import { Annotation, MessagesAnnotation } from "@langchain/langgraph";
import { SearchResult } from "@opencanvas/shared/types";

export const WebSearchGraphAnnotation = Annotation.Root({
  /**
   * The chat history to search the web for.
   * Will use the latest user message as the query.
   */
  ...MessagesAnnotation.spec,
  /**
   * The search query.
   */
  query: Annotation<string>,
  /**
   * The search results
   */
  webSearchResults: Annotation<SearchResult[]>,
  /**
   * Whether or not to search the web based on the user's latest message.
   */
  shouldSearch: Annotation<boolean>,
});

export type WebSearchState = typeof WebSearchGraphAnnotation.State;

File: agents-py/agents/src/web_search/state.py
from typing import List, Dict, Any
from pydantic import BaseModel, Field
from langchain_core.messages import BaseMessage
from shared.src.types import SearchResult

class WebSearchState(BaseModel):
    """State representation for Web Search graph"""
    
    messages: List[BaseMessage] = Field(
        default_factory=list,
        description="Chat history to search the web for. Uses latest user message as query."
    )
    query: str = Field(
        default="",
        description="The search query"
    )
    web_search_results: List[SearchResult] = Field(
        default_factory=list,
        description="The search results"
    )
    should_search: bool = Field(
        default=False,
        description="Whether to search the web based on the user's latest message"
    ) 